### ISSUE TITLE
Gallery image move and gallery.txt issue

### ISSUE THREAD

**Issue by sryu** [yuzoolcode/pulsecms-tracking#56]
I've posted a related bug before. In that case, it was regarding deletion of an image, and the gallery.txt not being properly updated. 

I see similar shortcomings (am using the 5.1 release version) when
1)  the file is renamed 
2) the file is moved to another directory
**especially when the said file has a caption entry in the existing gallery.txt file**.
The original and destination (the same in the case 1) directories' gallery.txt (if already existent) are not properly updated. 

This becomes problematic when used in conjunction with any stacks or custom code that extract information about the content of a gallery (what files are there, whether captions exist for each file or not, etc) and use that to present images (eg gallery stack, ...)

Since the .php codes that carry out 1) or 2) above should have access to the target dir and the gallery.txt files (if any), it should be fixable by calling a "update_gallery_text_file.php" that would perform 
1) updating file existence
2) updating caption, if any, for the affected entry in both directories
basically making sure that gallery.txt files in both directories to be up-to-date after those operations.

Without improvement, this becomes a very inconsistent experience for client at an editor level and potentially breaking front end gui design.

**Comment by ghost** [comment]
Thanks, related being:
https://github.com/yuzoolcode/pulsecms-tracking/issues/47

That was fixed:
https://github.com/yuzoolcode/pulsecms-tracking/issues/47#issuecomment-389761097

But this issue seems similar so will take a look, thanks for reporting

**Comment by ghost** [comment]
Fixed - Gallery image move and gallery.txt issue

The gallery text data for the image is now copied across. The unit tests have been updated too.

This is in 5.2BETA1
